netket/nn/fast_masked_linear.py

```diff
@@ -0,0 +1,499 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Tuple
+
+import flax
+from flax import linen as nn
+from jax import lax
+from jax import numpy as jnp
+
+from netket.nn.initializers import zeros
+from netket.nn.masked_linear import (
+    MaskedConv1D,
+    MaskedConv2D,
+    MaskedDense1D,
+    default_kernel_init,
+    wrap_kernel_init,
+)
+from netket.utils.types import Array, DType, NNInitFunc
+
+
+class FastMaskedDense1D(nn.Module):
+    """
+    1D linear transformation module with mask for fast autoregressive NN.
+    See :ref:`netket.nn.FastMaskedConv1D` for a brief explanation of fast autoregressive sampling.
+    TODO: FastMaskedDense1D does not support JIT yet
+
+    Attributes:
+      size: number of sites.
```

Comment:
can you document this in the same style as everywhere else? namely, under the attributes themselves?

Reply:
Ok, I'll also update `linear.py` and `masked_linear.py`. It seems that new code like `symmetric_linear.py` are already using this format.